packages/lix-sdk/src/database/kysely-plugin/serialize-jsonb-plugin.ts

```diff
@@ -1,82 +1,86 @@
 } from "kysely";
 
-export class SerializeJsonPlugin implements KyselyPlugin {
-	#parseJsonTransformer = new ParseJsonTransformer();
+export function SerializeJsonBPlugin(): KyselyPlugin {
+	const parseJsonTransformer = new SerializeJsonbTransformer();
 
-	transformQuery(args: PluginTransformQueryArgs): RootOperationNode {
-		if (
-			args.node.kind === "InsertQueryNode" ||
-			args.node.kind === "UpdateQueryNode"
-		) {
-			const result = this.#parseJsonTransformer.transformNode(args.node);
+	return {
+		transformResult: async (args) => args.result,
+		transformQuery(args) {
+			if (
+				args.node.kind === "InsertQueryNode" ||
+				args.node.kind === "UpdateQueryNode"
+			) {
+				const result = parseJsonTransformer.transformNode(args.node);
 
-			return result;
-		}
-		return args.node;
-	}
+				return result;
+			}
+			return args.node;
+		},
+	};
+}
 
-	async transformResult(
-		args: PluginTransformResultArgs,
-	): Promise<QueryResult<UnknownRow>> {
-		return args.result;
+class SerializeJsonbTransformer extends OperationNodeTransformer {
+	override transformOnConflict(node: OnConflictNode): OnConflictNode {
+		return super.transformOnConflict({
+			...node,
+			updates: node.updates?.map((updateItem) => {
+				if (updateItem.kind !== "ColumnUpdateNode") {
+					return updateItem;
+				}
+				return {
+					kind: "ColumnUpdateNode",
+					column: updateItem.column,
+					value: this.transformValue(
+						// @ts-expect-error - type mismatch
+						updateItem.value,
+					),
+				};
+			}),
+		});
 	}
-}
 
-class ParseJsonTransformer extends OperationNodeTransformer {
-	protected override transformValueList(node: ValueListNode): ValueListNode {
+	override transformValue(node: ValueNode): ValueNode {
+		const { value } = node;
+		const serializedValue = maybeSerializeJson(value);
+		if (value === serializedValue) {
+			return node;
+		}
+		// @ts-expect-error - type mismatch
+		return sql`jsonb(${serializedValue})`.toOperationNode();
+	}
+	/**
+	 * Transforms the value list node by replacing all JSON objects with `jsonb` function calls.
+	 */
+	override transformValueList(node: ValueListNode): ValueListNode {
 		return super.transformValueList({
 			...node,
 			values: node.values.map((listNodeItem) => {
 				if (listNodeItem.kind !== "ValueNode") {
 					return listNodeItem;
 				}
-
 				// @ts-expect-error - type mismatch
 				const { value } = listNodeItem;
-
 				const serializedValue = maybeSerializeJson(value);
 
 				if (value === serializedValue) {
 					return listNodeItem;
 				}
-
-				// TODO use jsonb. depends on parsing again
-				// https://github.com/opral/inlang-sdk/issues/132
-				return sql`json(${serializedValue})`.toOperationNode();
+				return sql`jsonb(${serializedValue})`.toOperationNode();
 			}),
 		});
 	}
 
 	/**
-	 * Directly transform the value to a json value.
-	 *
-	 * Input: { value: { a: 1 } }
-	 * Output: { value: "{ a: 1 }"" }
+	 * Why this function is needed or why this works remains a mystery.
```

Comment:
change the description to:
"changes PrimitiveValueListNode to ValueListNode to allow kysely to process objects

Reply:
implemented in commit 91d52b50b961dadbb9dddf0bc65196d8540c0647